pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -30,56 +40,121 @@ func init() {
 }
 
 type Service struct {
-	PluginManager      plugins.Manager     `inject:""`
-	HTTPClientProvider httpclient.Provider `inject:""`
-	Cfg                *setting.Cfg        `inject:""`
+	PluginManager        plugins.Manager       `inject:""`
+	HTTPClientProvider   httpclient.Provider   `inject:""`
+	Cfg                  *setting.Cfg          `inject:""`
+	BackendPluginManager backendplugin.Manager `inject:""`
 }
 
-func (s *Service) Init() error {
-	return nil
+type datasourceInfo struct {
+	AppInsightsAppId             string `json:"appInsightsAppId"`
+	AzureLogAnalyticsSameAs      bool   `json:"azureLogAnalyticsSameAs"`
+	ClientId                     string `json:"clientId"`
+	CloudName                    string `json:"cloudName"`
+	LogAnalyticsClientId         string `json:"logAnalyticsClientId"`
+	LogAnalyticsDefaultWorkspace string `json:"logAnalyticsDefaultWorkspace"`
+	LogAnalyticsSubscriptionId   string `json:"logAnalyticsSubscriptionId"`
+	LogAnalyticsTenantId         string `json:"logAnalyticsTenantId"`
+	SubscriptionId               string `json:"subscriptionId"`
+	TenantId                     string `json:"tenantId"`
+	AzureAuthType                string `json:"azureAuthType,omitempty"`
+
+	HTTPClient              *http.Client
+	URL                     string
+	JSONData                map[string]interface{}
+	DecryptedSecureJSONData map[string]string
+	DatasourceID            int64
 }
 
 // AzureMonitorExecutor executes queries for the Azure Monitor datasource - all four services
 type AzureMonitorExecutor struct {
-	httpClient    *http.Client
-	dsInfo        *models.DataSource
-	pluginManager plugins.Manager
-	cfg           *setting.Cfg
+	im                 instancemgmt.InstanceManager
+	pluginManager      plugins.Manager
+	cfg                *setting.Cfg
+	httpClientProvider httpclient.Provider
 }
 
-// NewAzureMonitorExecutor initializes a http client
-//nolint: staticcheck // plugins.DataPlugin deprecated
-func (s *Service) NewExecutor(dsInfo *models.DataSource) (plugins.DataPlugin, error) {
-	httpClient, err := dsInfo.GetHTTPClient(s.HTTPClientProvider)
-	if err != nil {
-		return nil, err
+func NewInstanceSettings() datasource.InstanceFactoryFunc {
+	return func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		opts, err := settings.HTTPClientOptions()
+		if err != nil {
+			return nil, err
+		}
+
+		client, err := sdkHTTPClient.New(opts)
```

Comment:
Shouldn't this be `httpClientProvider.New`? Like the `NewInstanceSettings` should take a `httpClientProvider httpclient.Provider` as argument?